src/features/positions/stores/transform/index.ts

```diff
@@ -0,0 +1,512 @@
+import {
+  PositionName,
+  type ListPositionsResponse,
+  type RainbowPositions,
+  type ProtocolGroup,
+  type Position,
+  type RainbowDapp,
+  type PortfolioItem,
+  type PositionToken,
+  type CategoryResult,
+  type RainbowUnderlyingAsset,
+  type PositionAsset,
+  type RainbowDeposit,
+  type RainbowPool,
+  type RainbowStake,
+  type RainbowBorrow,
+  type RainbowReward,
+} from '../../types';
+import type { PositionsParams } from '../fetcher';
+import { filterPositions } from './filter';
+import { sortPositions } from './sort';
+import { normalizeDappName } from './utils/dapp';
+import { calculatePositionTotals, calculateGrandTotals, calculateTotalValue, calculateTokenNativeDisplay } from './utils/totals';
+import { isConcentratedLiquidityProtocol, calculateLiquidityRangeStatus, calculateLiquidityAllocation } from './utils/lp';
+import { normalizeDate, normalizeDateTime } from './utils/date';
+import { convertAmountToNativeDisplay } from '@/helpers/utilities';
+import { NativeCurrencyKey } from '@/entities';
+
+// ============ Constants ====================================================== //
+
+const UNSUPPORTED_POSITION_TYPES: readonly (PositionName | string)[] = [
+  PositionName.PERPETUALS,
+  PositionName.OPTIONS_BUYER,
+  PositionName.OPTIONS_SELLER,
+  PositionName.INSURANCE_BUYER,
+  PositionName.INSURANCE_SELLER,
+];
+
+// ============ Helpers ======================================================== //
+
+/**
+ * Check if a DeBank position type should be filtered out
+ */
+function shouldFilterPositionType(positionName: PositionName | string): boolean {
+  return UNSUPPORTED_POSITION_TYPES.includes(positionName);
+}
+
+// ============ Asset Building ================================================= //
+
+/**
+ * Transform DeBank position tokens into underlying assets with native display values
+ */
+function transformUnderlyingAssets(tokens: PositionToken[] | undefined, currency: string): RainbowUnderlyingAsset[] {
+  if (!tokens || tokens.length === 0) {
+    return [];
+  }
+
+  return tokens
+    .map(token => {
+      if (!token.asset) return null;
+
+      return {
+        asset: {
+          ...token.asset,
+          chain_id: token.asset.chainId,
+          icon_url: token.asset.iconUrl,
+          // Normalize Go date format and filter out Go zero time
+          creationDate: normalizeDate(token.asset.creationDate),
+          price: token.asset.price
+            ? {
+                ...token.asset.price,
+                // Filter out Go zero time Date objects and convert to timestamp
+                changed_at: normalizeDateTime(token.asset.price.changedAt),
+                relative_change_24h: token.asset.price.relativeChange24h || 0,
+              }
+            : undefined,
+        } as PositionAsset,
```

Comment:
Does this not conform to `PositionAsset`?

If it does, I would remove the assertion or use `satisfies` if needed. If it doesn't, I'd fix that so we aren't masking type violations.